server_test.go

```diff
@@ -50,7 +72,15 @@ func (u *user) Logout() error {
 	return nil
 }
 
-func testServer(t *testing.T) (be *backend, s *smtp.Server, c net.Conn, scanner *bufio.Scanner) {
+type serverConfigureFunc func(*smtp.Server)
```

Comment:
All this to ensure `AuthDisabled` is set to the appropriate value *before* we send `EHLO`...